packages/backend/src/apps/tiles/actions/create-row/index.ts

```diff
@@ -0,0 +1,115 @@
+import { IRawAction } from '@plumber/types'
+
+import { generateStepError } from '@/helpers/generate-step-error'
+import { createTableRow } from '@/models/dynamodb/table-row/functions'
+import TableMetadata from '@/models/table-metadata'
+
+import { validateTileAccess } from '../../common/validate-tile-access'
+
+const action: IRawAction = {
+  name: 'Create row',
+  key: 'createTileRow',
+  description: 'Creates a new row in Tile',
+  arguments: [
+    {
+      label: 'Select Tile',
+      key: 'tableId',
+      type: 'dropdown' as const,
+      required: true,
+      variables: false,
+      description: 'Select the tile you want to create a row in.',
+      showOptionValue: false,
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'listTables',
+          },
+        ],
+      },
+    },
+    {
+      label: 'New row data',
+      key: 'rowData',
+      type: 'multirow' as const,
+      required: false,
+      subFields: [
+        {
+          placeholder: 'Column',
+          key: 'columnId',
+          type: 'dropdown' as const,
+          required: true,
+          variables: false,
+          showOptionValue: false,
+          source: {
+            type: 'query' as const,
+            name: 'getDynamicData' as const,
+            arguments: [
+              {
+                name: 'key',
+                value: 'listColumns',
+              },
+              {
+                name: 'parameters.tableId',
+                value: '{parameters.tableId}',
+              },
+            ],
+          },
+        },
+        {
+          placeholder: 'Value',
+          key: 'cellValue',
+          type: 'string' as const,
+          required: false,
+          variables: true,
+        },
+      ],
+    },
+  ],
+
+  async run($) {
+    const { tableId, rowData } = $.step.parameters as {
+      tableId: string
+      rowData: { columnId: string; cellValue: string }[]
+    }
+    await validateTileAccess($.flow?.userId, tableId as string)
+
+    const table = await TableMetadata.query().findById(tableId)
+
+    /**
+     * convert array to object
+     * [{columnId: 'abc', cellValue: 'test'}] => {abc: 'test'}
+     */
+    const rowDataObject = rowData.reduce((acc, { columnId, cellValue }) => {
```

Comment:
should we error out if we encounter duplicate columns.

Can copy from my [excel code](https://github.com/opengovsg/plumber/blob/9a065bcd79572df1219f7e10df32d3751350fd81/packages/backend/src/apps/m365-excel/actions/create-table-row/index.ts#L155) (FWIW I plan to update that with zod tbh, reasoning in #401 )